packages/coin-support-evm/src/operations/prepareTransaction/index.ts

```diff
@@ -34,42 +41,114 @@ const validateAddresses = (
   return outputAddressValidation;
 };
 
+const generateDataField = (params: {
+  contractAddress: string;
+  senderAddress: string;
+  receiverAddress: string;
+  amount: string;
+}) => {
+  try {
+    const Web3 = getCoinSupportWeb3Lib();
+    const web3 = new Web3();
+    const erc20Contract = new web3.eth.Contract(
+      erc20Abi,
+      params.contractAddress,
+      {
+        from: params.senderAddress,
+      },
+    );
+    let amount = '0';
+    const parsedAmountParam = new BigNumber(params.amount);
+    if (!parsedAmountParam.isNaN()) {
+      amount = parsedAmountParam.toString();
+    }
+
+    return (erc20Contract.methods.transfer as any)(
+      params.receiverAddress || params.senderAddress,
```

Comment:
I think we don't need OR operator here, since `receiverAddress` is a required parameter.
```suggestion
      params.receiverAddress,
```